### PR TITLE
UI: Display ModelContext in StatusBarFooter

### LINKED ISSUE #100: UI: Display current context
Right now, users only know which context they are in after executing a context-switching command, so displaying the context somewhere would help users orientate themselves.

### PR DESCRIPTION
**Pull request by rlrh**
#### Resolves #100 
Displays Model's current `ModelContext` in `StatusBarFooter`
#### Summary of changes:
- Make ModelContext observable
- Make StatusBarFooter update whenever ModelContextProperty changes
- Add friendly name field to ModelContext
- Rearrange StatusBarFooter elements
- Update StatusBarFooterTest and other affected tests

### CODE REVIEW

**Comment by epicfailname** [comment]
In ```AddressBookSystemTest```, there are assert functions for statusbar. We probably want to rename these asserts to also exclude the context!

File: src/main/java/seedu/address/logic/LogicManager.java [hunk 1]
```diff
@@ -116,4 +117,12 @@ public void setCommandResult(CommandResult commandResult) {
         model.setCommandResult(commandResult);
     }
 
+    @Override
+    public ReadOnlyProperty<ModelContext> contextProperty() {
+        return model.contextProperty();
+    }
+
+    @Override
+    public void setContext(ModelContext context) { model.setContext(context); }
```
Comment (epicfailname) [inline src/main/java/seedu/address/logic/LogicManager.java:126]: This function isn't currently used anywhere. Is it planned to be used in the future?

**Comment by thomastanck** [comment]
> Resolves #100 :tada:

File: src/main/java/seedu/address/ui/StatusBarFooter.java [hunk 2]
```diff
@@ -39,23 +41,35 @@
     @FXML
     private Label saveLocationStatus;
     @FXML
-    private Label targetCountStatus;
+    private Label contextEntryCountStatus;
 
-
-    public StatusBarFooter(Path saveLocation, ReadOnlyEntryBook addressBook, ObservableList<Entry> entryList) {
+    public StatusBarFooter(Path saveLocation,
+                           ReadOnlyEntryBook addressBook,
```
Comment (qjqqyy) [inline src/main/java/seedu/address/ui/StatusBarFooter.java:47]: Since we're here anyway might as well rename this also?

**Review by qjqqyy — APPROVED** [review]
LGTM, observer pattern is perfect for what we're trying to do here.

**Pull request merged**